agent/cmd/cuda-checkpoint-helper/main.cpp

```
@@ -0,0 +1,1546 @@
+
+bool RunHealthServer(daemon_protocol::OwnedUnixSocket *socket, int shutdown_fd,
+                     const daemon_protocol::OperationHealth *health,
+                     PersistentTargetContexts *persistent_contexts) {
+  std::vector<unsigned char> packet(daemon_protocol::kMaxRequestSize + 1);
+  for (;;) {
+    if (daemon_protocol::PollForInputOrStop(socket->fd(), shutdown_fd) == 0) {
+      return true;
+    }
+    const int accepted_fd =
+        accept4(socket->fd(), nullptr, nullptr, SOCK_CLOEXEC);
+    if (accepted_fd < 0) {
```

Clang (14.0.6)

[note] +2, including nesting penalty of 1, nesting level increased to 2

(clang)